src/scripts/main.js

```diff
@@ -1,3 +1,100 @@
 'use strict';
 
-// write code here
+const MIN_VALUE = 2;
+const MAX_VALUE = 10;
+
+const table = document.querySelector('.field');
+const rowRemove = document.querySelector('.remove-row');
+const rowAdd = document.querySelector('.append-row');
+const columnAdd = document.querySelector('.append-column');
+const columnRemove = document.querySelector('.remove-column');
+
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('.button');
+
+  if (!target) {
+    return;
+  }
+
+  changeTable(target);
+});
+
+function changeTable(elem) {
+  const addTableRow = elem.closest('.append-row');
+  const addTableColumn = elem.closest('.append-column');
+  const removeRow = elem.closest('.remove-row');
+  const removeColumn = elem.closest('.remove-column');
+
+  if (addTableRow) {
+    const cloneElem = table.rows[0].cloneNode(true);
+
+    table.append(cloneElem);
+
+    if (table.rows.length === MAX_VALUE) {
+      addAttribute(addTableRow);
+    }
+
+    check();
+  }
+
+  if (removeRow) {
+    table.rows[table.rows.length - 1].remove();
+
+    if (table.rows.length === MIN_VALUE) {
+      addAttribute(removeRow);
+    }
+
+    check();
+  }
+
+  if (addTableColumn) {
+    [...table.rows].forEach((element) => {
+      const td = document.createElement('td');
+
+      element.append(td);
+    });
+
+    if (table.rows[0].cells.length === MAX_VALUE) {
+      addAttribute(addTableColumn);
+    }
```

Comment:
When the number of columns reaches `MAX_VALUE`, the `addAttribute` function is called to disable the 'append-column' button. Ensure that this logic is correct and that the button is re-enabled when the number of columns decreases below `MAX_VALUE`.